fig.go

```diff
@@ -183,6 +220,36 @@ func stringToRegexpHookFunc() mapstructure.DecodeHookFunc {
 	}
 }
 
+// stringToStringUnmarshalerHook returns a DecodeHookFunc that executes a custom method which
+// satisfies the StringUnmarshaler interface on custom types.
+func stringToStringUnmarshalerHook() mapstructure.DecodeHookFunc {
+	return func(f reflect.Type, t reflect.Type, data interface{}) (interface{}, error) {
+		if f.Kind() != reflect.String {
+			return data, nil
+		}
+
+		ds, ok := data.(string)
+		if !ok {
+			return data, nil
+		}
+
+		if reflect.PointerTo(t).Implements(reflect.TypeOf((*StringUnmarshaler)(nil)).Elem()) {
```

Comment:
We're going to have to add similar logic in [setValue()](https://github.com/kkyr/fig/blob/9c6777ef40059185189f09cafbe365fb80f8f1ab/fig.go#L262) so that defaults and environment variables work as well.

Reply:
I'll look into it.